services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpointImpl.java

```diff
@@ -55,31 +55,62 @@
+            if (analysisContext == null)
+                throw new WindupException("AnalysisContext not provided and there's no default one for project #" + projectId + ".");
+        }
+        else {
+            // AnalysisContext came as input, save it as a new one and also as the default for the project.
+            analysisContext.setId(null);
+            analysisContext.setVersion(0);
+
+            //MigrationProject projectRef = entityManager.getReference(MigrationProject.class, projectId);
+            MigrationProject projectRef = entityManager.find(MigrationProject.class, projectId);
+            analysisContext.setMigrationProject(projectRef);
+
+            // If there is a default context, delete it and clone the new analysisContext to use it as a new default.
+            AnalysisContext defaultContext = this.analysisContextService.getDefaultProjectAnalysisContext(projectId);
+            if (null != defaultContext)
+            {
+                this.entityManager.remove(defaultContext);
+            }
+            // This one is the new default.
+            this.analysisContextService.create(analysisContext);
+
+            // This one will be used for execution.
+            analysisContext = this.analysisContextService.create(analysisContext);
+        }
+
+        WindupExecution execution = createAndTriggerExecution(analysisContext);
+        return execution;
+    }
+
+    @Override
     public WindupExecution executeWithContext(Long contextId)
     {
         AnalysisContext analysisContext = this.analysisContextService.get(contextId);
+        WindupExecution execution = createAndTriggerExecution(analysisContext);
+        return execution;
+    }
 
-        if (analysisContext.getApplications().size() == 0)
-        {
+    private WindupExecution createAndTriggerExecution(AnalysisContext analysisContext)
+    {
+        LOG.info("Creating execution. AnalysisContext: " + analysisContext);
+        if (analysisContext.getApplications().isEmpty())
             throw new BadRequestException("Cannot execute windup without selected applications");
-        }
-
-        for (RegisteredApplication application : analysisContext.getApplications())
-        {
-            application.setReportIndexPath(null);
-        }
 
         WindupExecution execution = new WindupExecution(analysisContext);
+        execution.setAnalysisContext(analysisContext);
+        execution.setProject(analysisContext.getMigrationProject());
         execution.setTimeStarted(new GregorianCalendar());
         execution.setState(ExecutionState.QUEUED);
-
         entityManager.persist(execution);
 
-        Path reportOutputPath = this.webPathUtil.createWindupReportOutputPath(
-                execution.getProject().getId().toString(),
-                execution.getId().toString()
-        );
-
+        Path reportOutputPath = this.webPathUtil.createWindupReportOutputPath("p" + execution.getProjectId(), "e" + execution.getId());
```

Comment:
1) It's only used once
2) If the code elsewhere relied on consistency of this method rather than execution's data, then that would be wrong.
3) The method docs is:
```
    /**
     * Creates an output path for the given project and group and name.
     */
    Path createWindupReportOutputPath(String projectPath, String name);
```
It doesn't say "projectId" and "executionId". Maybe we could change that, but as it is now, I am only changing project name and "group and name" (?).